presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -321,22 +324,31 @@ public long getMaxCombinedBytesPerRow()
     public void close()
             throws IOException
     {
-        orcDataSource.close();
-
-        if (writeChecksumBuilder.isPresent()) {
-            WriteChecksum actualChecksum = writeChecksumBuilder.get().build();
-            validateWrite(validation -> validation.getChecksum().getTotalRowCount() == actualChecksum.getTotalRowCount(), "Invalid row count");
-            List<Long> columnHashes = actualChecksum.getColumnHashes();
-            for (int i = 0; i < columnHashes.size(); i++) {
-                int columnIndex = i;
-                validateWrite(validation -> validation.getChecksum().getColumnHashes().get(columnIndex).equals(columnHashes.get(columnIndex)),
-                        "Invalid checksum for column %s", columnIndex);
+        try (Closer closer = Closer.create()) {
+            closer.register(() -> orcDataSource.close());
+            closer.register(() -> {
+                for (StreamReader column : streamReaders) {
+                    if (column != null) {
+                        column.close();
+                    }
+                }
+            });
+
+            if (writeChecksumBuilder.isPresent()) {
```

Comment:
Please don't wrap these if blocks in the try block.

Reply:
Sorry could you explain a bit why? I thought the purpose of putting the if blocks in the try block was that if some exceptions are thrown in these if blocks, the orcDataSource and streamReaders close() methods would always be called. However in this case, if we put the try block in front and don't wrap the if blocks you mentioned, it still work because the close() would be called before the if blocks are executed. Is this the reason you don't want to wrap these if blocks in the try block?